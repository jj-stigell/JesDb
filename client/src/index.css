@tailwind base;
@tailwind components;
@tailwind utilities;

@layer components {
  .link {
    @apply underline mx-1 text-blue-600 hover:text-blue-800 dark:text-blue-300 dark:hover:text-blue-600;
  }
  .title {
    @apply my-2 text-xl text-black font-extrabold dark:text-white;
  }
}

body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

.keyword {
  color: red;
}
