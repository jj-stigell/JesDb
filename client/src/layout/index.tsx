import React from 'react'
import { Outlet } from 'react-router-dom'

import Modal from './Modal'
import Menu from './Menu'
import Footer from './Footer'

export default function MainLayout (): React.JSX.Element {
  return (
    <>
      <div
        className="dark:bg-jesdark-900"
        style={{ minHeight: '100vh', display: 'flex', flexDirection: 'column' }}
      >
        <Modal />
        <Menu />
        <Outlet/>
      </div>
      <Footer />
    </>
  )
}
