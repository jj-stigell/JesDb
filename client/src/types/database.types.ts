// DO NOT EDIT THIS FILE
// TYPES ARE AUTOGENERATED FROM DATABASE SCHEMA

export type Json =
  | string
  | number
  | boolean
  | null
  | { [key: string]: Json | undefined }
  | Json[]

export interface Database {
  public: {
    Tables: {
      notes: {
        Row: {
          audio: string | null
          created_at: string
          external_id: number
          id: number
          image: string | null
          sentence: string
          sentence_furigana: string
          source: Database['public']['Enums']['media_source'] | null
          source_id: string | null
          start_time: number | null
          translation: string | null
        }
        Insert: {
          audio?: string | null
          created_at?: string
          external_id: number
          id?: number
          image?: string | null
          sentence: string
          sentence_furigana: string
          source?: Database['public']['Enums']['media_source'] | null
          source_id?: string | null
          start_time?: number | null
          translation?: string | null
        }
        Update: {
          audio?: string | null
          created_at?: string
          external_id?: number
          id?: number
          image?: string | null
          sentence?: string
          sentence_furigana?: string
          source?: Database['public']['Enums']['media_source'] | null
          source_id?: string | null
          start_time?: number | null
          translation?: string | null
        }
        Relationships: []
      }
      profiles: {
        Row: {
          action_type: Database['public']['Enums']['action_type']
          audio_field: string | null
          created_at: string
          deck_name: string
          expression_field: string | null
          id: number
          image_field: string | null
          name: string
          note_type: string
          reading_field: string | null
          translation_field: string | null
          user_id: string
        }
        Insert: {
          action_type: Database['public']['Enums']['action_type']
          audio_field?: string | null
          created_at?: string
          deck_name: string
          expression_field?: string | null
          id?: number
          image_field?: string | null
          name: string
          note_type: string
          reading_field?: string | null
          translation_field?: string | null
          user_id: string
        }
        Update: {
          action_type?: Database['public']['Enums']['action_type']
          audio_field?: string | null
          created_at?: string
          deck_name?: string
          expression_field?: string | null
          id?: number
          image_field?: string | null
          name?: string
          note_type?: string
          reading_field?: string | null
          translation_field?: string | null
          user_id?: string
        }
        Relationships: [
          {
            foreignKeyName: 'public_profile_user_id_fkey'
            columns: ['user_id']
            isOneToOne: false
            referencedRelation: 'users'
            referencedColumns: ['id']
          },
        ]
      }
      synonyms: {
        Row: {
          synonyms: string[] | null
          term: string
        }
        Insert: {
          synonyms?: string[] | null
          term: string
        }
        Update: {
          synonyms?: string[] | null
          term?: string
        }
        Relationships: []
      }
    }
    Views: {
      [_ in never]: never
    }
    Functions: {
      delete_user: {
        Args: Record<PropertyKey, never>
        Returns: undefined
      }
      search_sentences: {
        Args: {
          keyword: string
        }
        Returns: Array<{
          id: number
          sentence: string
          len: number
          sentence_furigana: string
          translation: string
          audio: string
          image: string
          start_time: number
          source_id: string
          highlighted_sentence: string
          highlighted_language: string
          sentence_furigana_highlight: string
        }>
      }
    }
    Enums: {
      action_type: 'addNote' | 'updateNote'
      media_source: 'youtube'
    }
    CompositeTypes: {
      [_ in never]: never
    }
  }
}

type PublicSchema = Database[Extract<keyof Database, 'public'>]

export type Tables<
  PublicTableNameOrOptions extends
  | keyof (PublicSchema['Tables'] & PublicSchema['Views'])
  | { schema: keyof Database },
  TableName extends PublicTableNameOrOptions extends { schema: keyof Database }
    ? keyof (Database[PublicTableNameOrOptions['schema']]['Tables'] &
    Database[PublicTableNameOrOptions['schema']]['Views'])
    : never = never,
> = PublicTableNameOrOptions extends { schema: keyof Database }
  ? (Database[PublicTableNameOrOptions['schema']]['Tables'] &
    Database[PublicTableNameOrOptions['schema']]['Views'])[TableName] extends {
      Row: infer R
    }
      ? R
      : never
  : PublicTableNameOrOptions extends keyof (PublicSchema['Tables'] &
  PublicSchema['Views'])
    ? (PublicSchema['Tables'] &
      PublicSchema['Views'])[PublicTableNameOrOptions] extends {
        Row: infer R
      }
        ? R
        : never
    : never

export type TablesInsert<
  PublicTableNameOrOptions extends
  | keyof PublicSchema['Tables']
  | { schema: keyof Database },
  TableName extends PublicTableNameOrOptions extends { schema: keyof Database }
    ? keyof Database[PublicTableNameOrOptions['schema']]['Tables']
    : never = never,
> = PublicTableNameOrOptions extends { schema: keyof Database }
  ? Database[PublicTableNameOrOptions['schema']]['Tables'][TableName] extends {
    Insert: infer I
  }
    ? I
    : never
  : PublicTableNameOrOptions extends keyof PublicSchema['Tables']
    ? PublicSchema['Tables'][PublicTableNameOrOptions] extends {
      Insert: infer I
    }
      ? I
      : never
    : never

export type TablesUpdate<
  PublicTableNameOrOptions extends
  | keyof PublicSchema['Tables']
  | { schema: keyof Database },
  TableName extends PublicTableNameOrOptions extends { schema: keyof Database }
    ? keyof Database[PublicTableNameOrOptions['schema']]['Tables']
    : never = never,
> = PublicTableNameOrOptions extends { schema: keyof Database }
  ? Database[PublicTableNameOrOptions['schema']]['Tables'][TableName] extends {
    Update: infer U
  }
    ? U
    : never
  : PublicTableNameOrOptions extends keyof PublicSchema['Tables']
    ? PublicSchema['Tables'][PublicTableNameOrOptions] extends {
      Update: infer U
    }
      ? U
      : never
    : never

export type Enums<
  PublicEnumNameOrOptions extends
  | keyof PublicSchema['Enums']
  | { schema: keyof Database },
  EnumName extends PublicEnumNameOrOptions extends { schema: keyof Database }
    ? keyof Database[PublicEnumNameOrOptions['schema']]['Enums']
    : never = never,
> = PublicEnumNameOrOptions extends { schema: keyof Database }
  ? Database[PublicEnumNameOrOptions['schema']]['Enums'][EnumName]
  : PublicEnumNameOrOptions extends keyof PublicSchema['Enums']
    ? PublicSchema['Enums'][PublicEnumNameOrOptions]
    : never
