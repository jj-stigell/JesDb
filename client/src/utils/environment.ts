/**
 * Current Node environment.
 * This variable indicates the environment in which the Node application is running.
 * Possible values: 'development', 'production', and 'test'.
 * Defaults to 'development' if the NODE_ENV environment variable is not set.
 */
export const NODE_ENV: string = process.env.REACT_APP_NODE_ENV ?? 'development'

/**
 * Supabase URL. This is the URL to your Supabase project.
 * You can find this key in your Supabase project settings.
 */
export const SUPABASE_URL: string = process.env.REACT_APP_SUPABASE_URL ?? ''

/**
 * Supabase API key. Only use public key here. Do not expose your private key.
 * This key is used to authenticate with the Supabase API.
 * You can find this key in your Supabase project settings.
 */
export const SUPABASE_KEY: string = process.env.REACT_APP_SUPABASE_KEY ?? ''

/**
 * Auth callback URL. This is the URL that the user will be redirected to after
 * they authenticate with Supabase.
 * This URL must be added to the list of "Redirect URIs" in your Supabase project settings.
 */
export const AUTH_CALLBACK_URL: string =
  process.env.REACT_APP_AUTH_CALLBACK_URL ?? 'http://localhost:3000/auth/callback'
