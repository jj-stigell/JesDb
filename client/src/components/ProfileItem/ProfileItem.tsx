import React from 'react'
import { Button } from 'flowbite-react'

import { Profile } from 'src/types'

interface IProfileItem {
  profile: Profile
  deleteProfile: () => void
}

export default function ProfileItem ({
  profile,
  deleteProfile
}: IProfileItem): React.JSX.Element {
  return (
    <li
      className={`
        flex justify-between items-center bg-gray-100
        dark:bg-jesdark-600 dark:border-jesdark-700 p-2 rounded-lg my-2
      `}
      role="listitem"
    >
      <div>
        <p
          id={`profile-name-${profile.name}`}
          className="text-lg dark:text-white"
          role="heading"
          aria-level={3}
        >
          {profile.name}
        </p>
        <p className="text-sm dark:text-gray-200">Target deck: {profile.deck_name}</p>
        <p className="text-sm dark:text-gray-200">Note type: {profile.note_type}</p>
        <p className="text-sm dark:text-gray-200">
          Action type: {profile.action_type === 'addNote' ? 'Add new note' : 'Update latest note'}
        </p>
      </div>
      <div>
        <Button
          color="failure"
          onClick={deleteProfile}
          aria-label={`Delete profile: ${profile.name}`}
        >
          Delete
        </Button>
      </div>
    </li>
  )
}
