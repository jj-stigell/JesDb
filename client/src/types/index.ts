import { Database } from './database.types'

/**
 * No params for deckNames, modelNamesAndIds, modelNames
 */
export type Action =
  'deckNames' |
  'updateNote' |
  'modelNamesAndIds' |
  'modelFieldNames' |
  'modelNames' |
  'findNotes' |
  'addNote' |
  'getNoteTags'

export interface UpdateNoteParams {
  note: {
    id: number
    fields: Record<string, string>
    tags: string[]
  }
}

export interface AddNoteParams {
  note: {
    deckName: string
    modelName: string
    fields: Record<string, string>
    options?: {
      allowDuplicate?: boolean
      duplicateScope?: string
      duplicateScopeOptions?: {
        deckName?: string
        checkChildren?: boolean
        checkAllModels?: boolean
      }
    }
    tags?: string[]
    audio?: Array<{
      url: string
      filename: string
      skipHash: string
      fields: string[]
    }>
    video?: Array<{
      url: string
      filename: string
      skipHash: string
      fields: string[]
    }>
    picture?: Array<{
      url: string
      filename: string
      skipHash: string
      fields: string[]
    }>
  }
}

export type NoteParams = UpdateNoteParams | AddNoteParams

interface ModelFieldNamesParams {
  modelName: string
}

interface FindNotesParams {
  query: string
}

interface GetNoteTagsParams {
  note: number
}

interface ActionParamsMap {
  deckNames: undefined
  updateNote: UpdateNoteParams
  modelNamesAndIds: undefined
  modelFieldNames: ModelFieldNamesParams
  modelNames: undefined
  findNotes: FindNotesParams
  addNote: AddNoteParams
  getNoteTags: GetNoteTagsParams
}

export interface ErrorResult {
  result: null
  error: string
}

interface ResultStringArray {
  result: string[]
  error: null
}

export interface FindNotes {
  result: number[]
  error: null
}

export interface AddNote {
  result: number
  error: null
}

export interface UpdateNoteResult {
  result: null
  error: null
}

interface ModelNamesAndIdsResult {
  result: Array<Record<string, number>>
  error: null
}

interface ActionResultMap {
  deckNames: ResultStringArray
  updateNote: UpdateNoteResult
  modelNamesAndIds: ModelNamesAndIdsResult
  modelFieldNames: ResultStringArray
  modelNames: ResultStringArray
  findNotes: FindNotes
  addNote: AddNote
  getNoteTags: ResultStringArray
}

export type ExecuteActionParams<T extends Action> =
  T extends keyof ActionParamsMap ? ActionParamsMap[T] : never

export type ExecuteActionReturnType<T extends Action> = ActionResultMap[T]

export interface Body<T extends Action> {
  action: Action
  version: number
  params?: ExecuteActionParams<T>
}

// Represents the note type that search function returns
export type Note = Database['public']['Functions']['search_sentences']['Returns'][0]

export type Profile = Database['public']['Tables']['profiles']['Row']

export type ActionType = Database['public']['Enums']['action_type']

export interface NewProfile extends Omit<Profile, 'id' | 'user_id' | 'created_at'> {}

export interface ExportResult {
  result: string | null
  error: string | null
}

export interface Filters {
  sort: 'relevance' | 'shortness' | 'longness'
  onlyTranslated: boolean
}
