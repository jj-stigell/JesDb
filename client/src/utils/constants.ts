import { Note, Profile } from 'src/types'

export const repoLink = 'https://github.com/jj-stigell/JesDb'

export const feedbackLink = 'https://forms.gle/SVNcSrujPHoy6UuG9'

export const ankiConnectDefaultAddress = 'http://127.0.0.1:8765'

// This one has highlighted_language set to 'ja' therefore it will have sentence_furigana_highlight
export const exampleNoteJa: Note = {
  id: 1863,
  sentence: '昔から 自動車の会社をやっていて',
  len: 16,
  sentence_furigana: '昔[むかし]から 自動車[じどうしゃ]の 会社[かいしゃ]をやっていて',
  translation: 'long time a go I had a car company',
  audio: `https://jynhfjtxpodsmjlnexrx.supabase.co/storage/v1/
  object/public/jesdb-media/gBumdOWWMhY/gBumdOWWMhY_s102_e104.mp3`,
  image: `https://jynhfjtxpodsmjlnexrx.supabase.co/storage/v1/
  object/public/jesdb-media/gBumdOWWMhY/gBumdOWWMhY_s102_e104.jpg`,
  start_time: 102,
  source_id: 'gBumdOWWMhY',
  highlighted_sentence: '昔から <span class="keyword">自動車</span>の会社をやっていて',
  highlighted_language: 'ja',
  sentence_furigana_highlight: '昔[むかし]から <span class="keyword">自動車</span>[じどうしゃ]の 会社[かいしゃ]をやっていて'
}

// This one has highlighted_language set to 'en' therefore sentence_furigana_highlight is empty
export const exampleNoteEn: Note = {
  id: 1135,
  sentence: 'それが怖くなったのでございます',
  len: 15,
  sentence_furigana: 'それが 怖[こわ]くなったのでございます',
  translation: 'I was scared of what was waiting for me.',
  audio: `https://jynhfjtxpodsmjlnexrx.supabase.co/storage/v1/
  object/public/jesdb-media/ylEimFz8KOo/ylEimFz8KOo_s159_e162.mp3`,
  image: `https://jynhfjtxpodsmjlnexrx.supabase.co/storage/v1/
  object/public/jesdb-media/ylEimFz8KOo/ylEimFz8KOo_s159_e162.jpg`,
  start_time: 159,
  source_id: 'ylEimFz8KOo',
  highlighted_sentence: 'I was <span class="keyword">scared</span> of what was waiting for me.',
  highlighted_language: 'en',
  sentence_furigana_highlight: ''
}

export const exampleProfile: Profile = {
  id: 1,
  name: 'Test Profile',
  deck_name: 'Test Deck',
  note_type: 'Basic',
  action_type: 'addNote',
  expression_field: 'expression',
  reading_field: 'reading',
  translation_field: 'translation',
  audio_field: 'audio',
  image_field: 'image',
  created_at: '2021-09-01T00:00:00.000Z',
  user_id: '1af8625c-0771-4c53-9851-42274e8e2385'
}
