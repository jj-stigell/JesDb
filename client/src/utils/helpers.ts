import { NoteParams, Note, Profile } from 'src/types'

export function getFileName (url: string): string {
  const regex = /[^/]+\.(mp3|jpg|png)$/
  const match = url.match(regex)

  if (match !== null) {
    return match[0]
  } else {
    return `${(+new Date()).toString(36)}.${url.split('.').pop() ?? 'file'}`
  }
}

export function scrollToTop (): void {
  window.scrollTo({
    top: 0,
    behavior: 'smooth'
  })
}

export function constructParams (
  example: Note,
  profile: Profile
): NoteParams {
  const fields: any = {}
  const audio: any = {}
  const picture: any = {}

  if (profile.expression_field !== 'omit' && profile.expression_field !== null) {
    fields[profile.expression_field] = example.sentence
  }

  if (profile.reading_field !== 'omit' && profile.reading_field !== null) {
    fields[profile.reading_field] = example.sentence_furigana
  }

  if (profile.translation_field !== 'omit' && profile.translation_field !== null) {
    fields[profile.translation_field] = example.translation
  }

  if (profile.audio_field !== 'omit' && example.audio !== null && profile.audio_field !== null) {
    fields[profile.audio_field] = ''
    audio.url = example.audio
    audio.filename = getFileName(example.audio)
    audio.fields = [profile.audio_field]
  }

  if (profile.image_field !== 'omit' && example.image !== null && profile.image_field !== null) {
    fields[profile.image_field] = ''
    picture.url = example.image
    picture.filename = getFileName(example.image)
    picture.fields = [profile.image_field]
  }

  const params: any = {
    note: {
      fields,
      tags: ['JesDb', example.source_id],
      audio: [audio],
      picture: [picture]
    }
  }

  return params
}

export function parseRuby (input: string): string {
  const regex = /(^|\s|<[^>]+>)([^[<\s]+)\[([^\]]+)\]/g
  return input.replace(regex, (match, p1, kanji: string, furigana: string) => {
    const prefix: string = p1.trim().length > 0 ? p1 : ''
    return `${prefix}<ruby>${kanji}<rt>${furigana}</rt></ruby>`
  })
}
