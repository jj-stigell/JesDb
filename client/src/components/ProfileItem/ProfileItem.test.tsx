import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'

import ProfileItem from './ProfileItem'
import { Profile } from 'src/types'
import { exampleProfile } from '../../utils/constants'

describe('ProfileItem Component', () => {
  const mockDeleteProfile = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('renders profile information correctly', () => {
    const { asFragment } =
      render(<ProfileItem profile={exampleProfile} deleteProfile={mockDeleteProfile} />)

    expect(asFragment()).toMatchSnapshot()
    expect(screen.getByText(exampleProfile.name)).toBeInTheDocument()
    expect(screen.getByText(`Target deck: ${exampleProfile.deck_name}`)).toBeInTheDocument()
    expect(screen.getByText(`Note type: ${exampleProfile.note_type}`)).toBeInTheDocument()
    expect(screen.getByText('Action type: Add new note')).toBeInTheDocument()
  })

  it('calls deleteProfile when the delete button is clicked', () => {
    render(<ProfileItem profile={exampleProfile} deleteProfile={mockDeleteProfile} />)
    fireEvent.click(screen.getByText('Delete'))

    expect(mockDeleteProfile).toHaveBeenCalledTimes(1)
  })

  it('renders "Update latest note" when actionType is updateNote', () => {
    const updatedProfile: Profile = {
      ...exampleProfile,
      action_type: 'updateNote'
    }
    const { asFragment } =
      render(<ProfileItem profile={updatedProfile} deleteProfile={mockDeleteProfile} />)

    expect(asFragment()).toMatchSnapshot()
    expect(screen.getByText('Action type: Update latest note')).toBeInTheDocument()
  })
})
