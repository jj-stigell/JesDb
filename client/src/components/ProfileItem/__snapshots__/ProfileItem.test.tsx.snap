// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileItem Component renders "Update latest note" when actionType is updateNote 1`] = `
<DocumentFragment>
  <li
    class="
        flex justify-between items-center bg-gray-100
        dark:bg-jesdark-600 dark:border-jesdark-700 p-2 rounded-lg my-2
      "
    role="listitem"
  >
    <div>
      <p
        aria-level="3"
        class="text-lg dark:text-white"
        id="profile-name-Test Profile"
        role="heading"
      >
        Test Profile
      </p>
      <p
        class="text-sm dark:text-gray-200"
      >
        Target deck: Test Deck
      </p>
      <p
        class="text-sm dark:text-gray-200"
      >
        Note type: Basic
      </p>
      <p
        class="text-sm dark:text-gray-200"
      >
        Action type: Update latest note
      </p>
    </div>
    <div>
      <button
        aria-label="Delete profile: Test Profile"
        class="group relative flex items-stretch justify-center p-0.5 text-center font-medium transition-[color,background-color,border-color,text-decoration-color,fill,stroke,box-shadow] focus:z-10 focus:outline-none border border-transparent bg-red-700 text-white focus:ring-4 focus:ring-red-300 enabled:hover:bg-red-800 dark:bg-red-600 dark:focus:ring-red-900 dark:enabled:hover:bg-red-700 rounded-lg"
        type="button"
      >
        <span
          class="flex items-stretch transition-all duration-200 rounded-md px-4 py-2 text-sm"
        >
          Delete
        </span>
      </button>
    </div>
  </li>
</DocumentFragment>
`;

exports[`ProfileItem Component renders profile information correctly 1`] = `
<DocumentFragment>
  <li
    class="
        flex justify-between items-center bg-gray-100
        dark:bg-jesdark-600 dark:border-jesdark-700 p-2 rounded-lg my-2
      "
    role="listitem"
  >
    <div>
      <p
        aria-level="3"
        class="text-lg dark:text-white"
        id="profile-name-Test Profile"
        role="heading"
      >
        Test Profile
      </p>
      <p
        class="text-sm dark:text-gray-200"
      >
        Target deck: Test Deck
      </p>
      <p
        class="text-sm dark:text-gray-200"
      >
        Note type: Basic
      </p>
      <p
        class="text-sm dark:text-gray-200"
      >
        Action type: Add new note
      </p>
    </div>
    <div>
      <button
        aria-label="Delete profile: Test Profile"
        class="group relative flex items-stretch justify-center p-0.5 text-center font-medium transition-[color,background-color,border-color,text-decoration-color,fill,stroke,box-shadow] focus:z-10 focus:outline-none border border-transparent bg-red-700 text-white focus:ring-4 focus:ring-red-300 enabled:hover:bg-red-800 dark:bg-red-600 dark:focus:ring-red-900 dark:enabled:hover:bg-red-700 rounded-lg"
        type="button"
      >
        <span
          class="flex items-stretch transition-all duration-200 rounded-md px-4 py-2 text-sm"
        >
          Delete
        </span>
      </button>
    </div>
  </li>
</DocumentFragment>
`;
